var/spack/repos/builtin/packages/intel/package.py

```diff
@@ -0,0 +1,113 @@
+from spack import *
+import os
+
+class Intel(Package):
+    """Intel Compilers.
+
+    Note: You will have to add add the downolad file to a
+    mirror so that Spack can find it. For instructions on how to set up a
+    mirror, see http://software.llnl.gov/spack/mirrors.html"""
+
+    homepage = "http://www.intel.com"
+
+    # TODO: can also try the online installer (will download files on demand)
+    version('composer.2016.2', '1133fb831312eb519f7da897fec223fa',
+            url="file://%s/parallel_studio_xe_2016_composer_edition_update2.tgz" % os.getcwd())
+    version('professional.2016.2', '70be832f2d34c9bf596a5e99d5f2d832',
+            url="file://%s/parallel_studio_xe_2016_update2.tgz" % os.getcwd())
+    version('cluster.2016.2', '70be832f2d34c9bf596a5e99d5f2d832',
+            url="file://%s/parallel_studio_xe_2016_update2.tgz" % os.getcwd())
+
+    variant('rpath', default=True, description="Add rpath to .cfg files")
+    variant('all', default=False, description="Install all files associated with the requested edition")
+    variant('mpi', default=True, description="Install the Intel MPI library and ITAC tool")
+    variant('mkl', default=True, description="Install the Intel MKL library")
+    variant('daal', default=True, description="Install the Intel DAAL libraries")
+    variant('ipp', default=True, description="Install the Intel IPP libraries")
+    variant('tools', default=True, description="Install the Intel Advisor, VTune Amplifier, and Inspector tools")
+
+    provides('mpi', when='@cluster:+mpi')
+    provides('mkl', when='+mkl')
+    provides('daal', when='+daal')
+    provides('ipp', when='+ipp')
+
+    def install(self, spec, prefix):
+        
+        # remove the installation DB, otherwise it will try to install into location of other Intel builds
+        try:
+            os.remove(os.path.join(os.environ["HOME"], "intel", "intel_sdp_products.db"))
+        except OSError:
+            pass # if the file does not exist
+
+        # TODO: Update to use pull request 558
+        license_path = "/usr/local/etc/license.client.intel.lic"
+
+        base_components = "ALL" # when in doubt, install everything
```

Comment:
let see if I can remember my own reasoning... I think I set base_components to "ALL" just in case we encounter versions that aren't currently covered. However, I didn't want the +all variant to be default because it made it harder to disable components (i.e., the user would have to explicitly specify ~all~ipp instead of just ~ipp). Let me know if you think there is a better way to handle all of this.